### ISSUE TITLE
Object actor field is list not string

### ISSUE THREAD

**Issue by kaniini** [Plume-org/Plume#82]
object actor field for blog posts is: `"actor": ["https://baptiste.gelez.xyz/@/kaniini/"]`

should be: `"actor": "https://baptiste.gelez.xyz/@/kaniini/"`

**Comment by kaniini** [comment]
this causes problems when relaying the object by reference:

```
07:18:17.457 [info] Handling incoming AP activity
07:18:17.460 [info] Fetching https://baptiste.gelez.xyz/~/KaniiniTestBlog/current-status-of-plume-and-pleroma-federation/ via AP
07:18:18.047 [error] Task #PID<0.6438.0> started from Pleroma.Web.Federator terminating
** (Ecto.Query.CastError) deps/ecto/lib/ecto/repo/queryable.ex:357: value `["https://baptiste.gelez.xyz/@/kaniini/"]` in `where` cannot be cast to type :string in query:

from u in Pleroma.User,
  where: u.ap_id == ^["https://baptiste.gelez.xyz/@/kaniini/"],
  select: u

    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1397: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
Function: #Function<11.65409860/0 in Pleroma.Web.Federator.maybe_start_job/2>
    Args: []
```

**Comment by BaptisteGelez** [comment]
Are you sure this is the `actor` field which is causing this problem? Because I'm almost sure the following function is not creating an array, but a string: https://github.com/Plume-org/Plume/blob/master/plume-models/src/posts.rs#L172 (otherwise it would have been called `set_actor_link_vec`).

Maybe it is caused by the `attributedTo` field, which is an array?

**Comment by redmatrix** [comment]
It's entirely legal for actors to be arrays in ActivityStreams (and therefore in ActivityPub). Multiple actors are allowed and there is an example of this in one of the specs.

**Comment by fdb-hiroshima** [comment]
I've tested against today code as it is difficult to test against what it was back in June (database migrations happened, back then there was no sqlite support...).
Anyway, I'm getting this json (some part are redacted for concision) : 
```json
{
   "@context":[...],
   "actor":"https://plume-one.local/@/Trinity/",
   "cc":[ ],
   "id":"https://plume-one.local/~/Test/is-actor-an-array/activity",
   "object":{
      "attributedTo":[
         "https://plume-one.local/@/Trinity/",
         "https://plume-one.local/~/Test/"
      ],
      "cc":[

      ],
      "content":"<p>no content for now</p>\n",
      "id":"https://plume-one.local/~/Test/is-actor-an-array/",
      "license":"CC-0",
      "name":"Is actor an array?",
      "published":"2018-12-13T18:40:57.154461Z",
      "source":{
         "content":"no content for now",
         "mediaType":"text/markdown"
      },
      "summary":"",
      "tag":[

      ],
      "to":[
         "https://www.w3.org/ns/activitystreams#Public"
      ],
      "type":"Article",
      "url":"https://plume-one.local/~/Test/is-actor-an-array/"
   },
   "signature":{
      "created":"2018-12-13T18:40:57.744910813+00:00",
      "creator":"https://plume-one.local/@/Trinity/#main-key",
      "signatureValue": ...,
      "type":"RsaSignature2017"
   },
   "to":[
      "https://www.w3.org/ns/activitystreams#Public"
   ],
   "type":"Create"
}
```
`actor` is currently a string, not an array. I *think* this issue can be closed, if it was a bug at the time, it's now fixed.

**Comment by BaptisteGelez** [comment]
I'm going to close this issue then. @kaniini if you ever encounter it again, tell us, we will try to find what is wrong.
